### PR TITLE
Simplify and unify EnumCalendars interop

### PR DESCRIPTION
**Pull request by jkotas**
Windows was not allocating GCHandles, switch Unix to do the same.

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 Checked corefx_baseline
@dotnet-bot test Ubuntu x64 Checked corefx_baseline

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test  please

File: src/mscorlib/shared/System/Globalization/CalendarData.Unix.cs [hunk 1]
```diff
@@ -290,24 +293,16 @@ internal static bool EnumCalendarInfo(string localeName, CalendarId calendarId,
             return result;
         }
 
-        private static bool EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, CallbackContext callbackContext)
+        private static unsafe bool EnumCalendarInfo(string localeName, CalendarId calendarId, CalendarDataType dataType, ref EnumCalendarsData callbackContext)
```
Comment (krwq) [inline src/mscorlib/shared/System/Globalization/CalendarData.Unix.cs]: Does this need unsafe?

**Comment by jkotas** [comment]
@krwq Thanks!

**Pull request merged**